go.mod

```diff
@@ -15,16 +15,16 @@ require (
 	github.com/aws/aws-sdk-go-v2 v1.17.8
 	github.com/aws/aws-sdk-go-v2/config v1.18.16
 	github.com/aws/aws-sdk-go-v2/service/ecr v1.18.9
-	github.com/containerd/console v1.0.3
+	github.com/containerd/console v1.0.3 // indirect
 	github.com/containerd/containerd v1.7.2 // indirect
-	github.com/docker/cli v24.0.2+incompatible
+	github.com/docker/cli v24.0.5+incompatible
 	github.com/docker/distribution v2.8.2+incompatible
-	github.com/docker/docker v24.0.0-rc.2.0.20230706181717-98d3da79ef9c+incompatible
+	github.com/docker/docker v24.0.0-rc.2.0.20230723142919-afd4805278b4+incompatible
 	github.com/go-logr/logr v1.2.4
 	github.com/go-logr/zapr v1.2.3
 	github.com/gofrs/flock v0.8.1 // indirect
 	github.com/h2non/filetype v1.1.3
-	github.com/moby/buildkit v0.12.0
+	github.com/moby/buildkit v0.12.1-0.20230831174303-e2dfabdebce2
```

Comment:
you keep diving too deep. If you look at how NewPrinter works, it will do the right thing given 0.12.2 or master. The whole point is that it does the right thing and is updated to match the changing APIs.